src/connection.c

```diff
@@ -503,6 +508,52 @@ free_resolv_cb_data(struct resolv_cb_data *cb_data) {
 }
 
 static void
+send_proxy_header(int clientfd, const struct Connection *con) {
```

Comment:
Yeah, you are pushing the proxy header on to buffer very early. Although, I see why you are doing this since otherwise the buffer is filled with the client request. I'm not opposed to populating each connection with a proxy header, but there is a very strong case for selectively including this after the backend selection is made (i.e. We would consume the proxy header off the buffer if that backend wasn't using the proxy protocol).

Looking more carefully, you do not need to pass `clientfd` at all, since the sockaddr is available via `con->listener->address`, eliminating the need to call `getsockname()`.